File: KeenASR_SpeechRecognizer/ViewController.m
//
//  ViewController.m
//  KeenASR_SpeechRecognizer
//
//  Created by Cyril Garcia on 6/26/19.
//  Copyright © 2019 Cyril Garcia. All rights reserved.
//

#import "ViewController.h"
#import "SpeechRecognizerModel.h"
@import MediaPlayer;

@interface ViewController ()
@property (nonatomic, strong) IBOutlet UILabel *responseLabel;
@property (nonatomic, strong) IBOutlet UILabel *confidenceLebel;

@property (nonatomic, weak) KIOSRecognizer *recognizer;
@property (nonatomic, strong) SpeechRecognizerModel *speechRecognizerModel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.speechRecognizerModel = [[SpeechRecognizerModel alloc] init];
    
    self.speechRecognizerModel.delegate = self;
    
    [self.speechRecognizerModel prepMic];
    
    NSArray *phrases = @[@"yes", @"no", @"sure", @"not now", @"go ahead", @"later", @"alright"];
    [self.speechRecognizerModel createDecodingGraph:phrases];
    
}

-(IBAction)startListening:(id)sender {
    [self.speechRecognizerModel startListening];
}

- (void)intent:(int)intent :(NSNumber *)confidenceLevel {
    if (intent == 1) {
        NSLog(@"True");
    } else {
        NSLog(@"False");
    }
}

- (void)recognized:(NSString *)text {
    NSLog(@"Recognized text:%@",text);
}
@end

File: Hello_OpenEars/ViewController.swift
//
//  ViewController.swift
//  Hello_OpenEars
//
//  Created by Cyril Garcia on 6/26/19.
//  Copyright © 2019 Cyril Garcia. All rights reserved.
//

import UIKit

class ViewController: UIViewController, SpeechRecognizerModelDelegate {
  
    @IBOutlet var intentLabel: UILabel!
    @IBOutlet var recognizedSpeechLabel: UILabel!
    @IBOutlet var startListeningButton: UIButton!
    
    private var speechRecognizerModel = SpeechRecognizerModel()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        speechRecognizerModel.delegate = self
        speechRecognizerModel.prepModel(with: ["yes","sure","okay","alright","no","not now","no thanks","later"])
    }
    
    @IBAction func startListening() {
        startListeningButton.isHidden = true
        speechRecognizerModel.startListening()
    }
    
    @IBAction func stopListening() {
        speechRecognizerModel.stopListening()
    }
    
    func intent(_ intent: Bool?, _ recognizedWord: String) {
        print("Recognized Word: \(recognizedWord)     Intent: \(intent)")
        intentLabel.text = "\(intent)"
        recognizedSpeechLabel.text = recognizedWord
    }
    
    func didFinishDetectingSpeech() {
        print("Done")
    }
    

}
